eclipse-theia/theia
# Add inversify lifecycle to frontend preload script

#### What it does

Currently, the browser preload scripts has 3 tasks:
* Load the translations for the selected locale
* Load the backend OS to correctly display stuff like line endings
* Set the background of the application based on the last selected theme

However, these tasks are hardcoded and adopters cannot add new scripts that are supposed to be executed during the preload phase of the frontend application. This change adds an inversify lifecycle to the aforementioned frontend preload scripts and a new contribution point to the `package.json` called `frontendPreload`. This allows to define new `PreloadContribution` items.

It also changes the way communication within the preload script is being performed. Instead of using `fetch`, we use the normal rpc messaging system via `WebSocketConnectionProvider`.

As a drive by, also:
Closes https://github.com/eclipse-theia/theia/issues/12670
Closes https://github.com/eclipse-theia/theia/issues/6675

#### How to test

1. Confirm that the tasks outlined above still work as expected.
2. Remove the `frontendPreload` from `@theia/core/package.json`. Assert that the preload module is removed from the generated `index.js` file.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
